cmake/FindScotch.cmake

```diff
@@ -0,0 +1,68 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Find Scotch: https://www.labri.fr/perso/pelegrin/scotch/
+# If not in one of the default paths specify -D SCOTCH_ROOT=/path/to/Scotch
+# to search there as well.
+
+if(NOT SCOTCH_ROOT)
+  # Need to set to empty to avoid warnings with --warn-uninitialized
+  set(SCOTCH_ROOT "")
+  set(SCOTCH_ROOT $ENV{SCOTCH_ROOT})
+endif()
+
+# Find the Scotch include directory
+find_path(SCOTCH_INCLUDE_DIR scotch.h
+  PATH_SUFFIXES include
+  HINTS ${SCOTCH_ROOT})
+
+find_library(SCOTCH_LIB
+  NAMES scotch
+  PATH_SUFFIXES lib64 lib
+  HINTS ${SCOTCH_ROOT})
+
+find_library(SCOTCH_ERR_LIB
+  NAMES scotcherr
+  PATH_SUFFIXES lib64 lib
+  HINTS ${SCOTCH_ROOT})
+
+set(SCOTCH_LIBRARIES "${SCOTCH_LIB};${SCOTCH_ERR_LIB}" CACHE INTERNAL "Blah")
```

Comment:
Is it necessary to set this as `CACHE`? I'm asking because the `Blah` suggests this may have been something you just tested :)

Reply:
Don't remember. Removing it doesn't seem to do any harm :shrug: